### PR TITLE
Improve documentation of PodController and relating interfaces

### PR DESCRIPTION
**Pull request by e-sumin**
## Change Overview

Improved documentation for PodController and related interfaces. 

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [x] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kube/pod_controller.go [hunk 1]
```diff
@@ -223,6 +229,8 @@ func (p *podController) getContainerName() string {
 	return p.pod.Spec.Containers[0].Name
 }
 
+// StreamPodLogs returns io.ReadCloser which could be used to receive logs from pod
+// container will be decided basing on result of getContainerName function.
```
Comment (pavannd1) [inline pkg/kube/pod_controller.go:233]:
```suggestion
// Container will be decided based on the result of getContainerName function.
```

File: pkg/kube/pod_controller.go [hunk 2]
```diff
@@ -231,6 +239,10 @@ func (p *podController) StreamPodLogs(ctx context.Context) (io.ReadCloser, error
 	return StreamPodLogs(ctx, p.cli, p.pod.Namespace, p.pod.Name, p.getContainerName())
 }
 
+// GetCommandExecutor returns PodCommandExecutor instance which is configured to execute commands within pod controlled
+// by this controller.
+// If pod is not created or not ready yet, it will fail with an appropriate error.
+// container will be decided basing on result of getContainerName function
```
Comment (pavannd1) [inline pkg/kube/pod_controller.go:245]:
```suggestion
// Container will be decided based on the result of getContainerName function
```

File: pkg/kube/pod_controller.go [hunk 3]
```diff
@@ -254,6 +266,9 @@ func (p *podController) GetCommandExecutor() (PodCommandExecutor, error) {
 	return pce, nil
 }
 
+// GetFileWriter returns PodFileWriter instance which is configured to write file to pod controlled by this controller.
+// If pod is not created or not ready yet, it will fail with an appropriate error.
+// container will be decided basing on result of getContainerName function
```
Comment (pavannd1) [inline pkg/kube/pod_controller.go:271]:
```suggestion
// Container will be decided based on the result of getContainerName function
```

**Review by pavannd1 — APPROVED** [review]
Just minor suggestions. Looks good otherwise.

**Comment by e-sumin** [comment]
Thank you, suggestions were accepted.

**Pull request merged**